fractal_server/types/__init__.py

```diff
@@ -53,6 +56,7 @@
 
 ZarrDirStr = Annotated[
     NonEmptyStr,
+    AfterValidator(val_canonical_path),
```

Comment:
Since this validator is a slightly more custom one, let's make sure that it is called last.
That is, `relative-path/../xxx` should fail as a non-absolute path.